### PR TITLE
First version of the j-pet-format-converter to DICOM

### PR DESCRIPTION
**Pull request by Rav2**
Basic functionality created.
Possibility to convert from raw binary image encoded using integer values added.
Some functions to create dummy files added.

### CODE REVIEW

File: binary2DICOM.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+    frames = 0
+    bytes_per_pix = 2
+    is_signed = False
+    is_float = False
+    byte_order = ''
+    result = True
+    add_arg = 0
+    conv = True
+    # Checking user's input
+    sys_len = len(sys.argv)
```
Comment (wkrzemien) [inline binary2DICOM.py]: Have a look at https://docs.python.org/3/library/argparse.html to handle command line arguments

File: binary2DICOM.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
+# binary2DICOM converter
```
Comment (wkrzemien) [inline binary2DICOM.py]: The first line should be:
#!/usr/bin/env python

File: binary2DICOM.py [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    """
+    A naive parser for DICOM meta-data.
+    :param filename: Contains meta-data in a format: <group tag> , <element tag> , <VR>, <value> Commas are delimiters.
+    :return: An array N x 3, where N is the number of correctly parsed lines.
+    """
+    meta_arr = []
+    line_no = 0
+    lines_with_data = 0
+    with open(filename) as f:
+        for line in f:
```
Comment (wkrzemien) [inline binary2DICOM.py]: you can use
 for line_no, line  in enumerate(f)

File: binary2DICOM.py [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+
+
+def convert(in_file, out_file, width, height, frames, bytes_per_pix=2, is_signed=False, is_float=False, byte_order='', meta_file=''):
+    meta_arr = []
+    try:
+        if meta_file != '':
+            meta_arr = read_meta(meta_file)
+    except FileNotFoundError as fnfe:
+        print("WARNING: Meta-data was not loaded, because the file was not found: "+str(fnfe))
+        print("I will now proceed to reading an image data.")
```
Comment (wkrzemien) [inline binary2DICOM.py]: there is no point to go further if the exception is rised, you should return from function here

File: binary2DICOM.py [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+    image_arr = read_binary(in_file, width, height, frames, bytes_per_pix, is_signed, is_float, byte_order)
+    if_success = write_dicom(out_file, image_arr, bytes_per_pix, is_signed, is_float, meta_arr)
+    return if_success
+
+if __name__ == "__main__":
+    in_file = ''
```
Comment (wkrzemien) [inline binary2DICOM.py]: I would put all those arguments in some dict

File: binary2DICOM.py [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+        ds.PixelData = pixel_array.tostring()
+
+        ds.save_as(filename.replace('.dcm', '') + '.dcm')
+        print('Writing to DICOM file is complete!')
+        return True
+    except ValueError as ve:
+        print(err_str+'ValueError '+str(ve))
+        return False
+    except FileExistsError as fe:
+        print(err_str+'FileExistsError '+str(fe))
```
Comment (wkrzemien) [inline binary2DICOM.py]: here you dont return anything?

File: binary2DICOM.py (hunk 2 above)
Comment (wkrzemien) [inline binary2DICOM.py]: In general you should decide how do you handle errors, or you return values or use exceptions

File: binary2DICOM.py [hunk 7]
```diff
@@ -0,0 +1,364 @@
...
+def test2d():
+    """
+    Function creates a dummy file with a 2D gradient.
+    :return: True
+    """
+    print('Making 2d image filled with gradient squares.')
+    x = np.arange(16).reshape(16,1) # tworzymy gradient 1D
+    pixel_array = (x + x.T) * 16 # z tablicy robimy gradient 2D i wygladzamy
+    pixel_array = np.tile(pixel_array,(4,4)) # powielamy 4 razy wzdluz X i Y
+    write_dicom('pretty2d.dcm', pixel_array)
```
Comment (wkrzemien) [inline binary2DICOM.py]: return write_dicom(//)

No point of returning always true

File: binary2DICOM.py [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+    """
+    Function creates a dummy file with a 2D gradient, repeated on 4 slices.
+    :return: 
+    """
+    print('Making 3d image filled with gradient squares.')
+    x = np.arange(16).reshape(16, 1)  # tworzymy gradient 1D
+    pixel_array = (x + x.T) * 16  # z tablicy robimy gradient 2D i wygladzamy
+    pixel_array = np.tile(pixel_array, (4, 4, 4))
+    write_dicom('pretty3d.dcm', pixel_array)
+    return True
```
Comment (wkrzemien) [inline binary2DICOM.py]: not needed

File: binary2DICOM.py [hunk 9]
```diff
@@ -0,0 +1,364 @@
...
+        if meta_file != '':
+            meta_arr = read_meta(meta_file)
+    except FileNotFoundError as fnfe:
+        print("WARNING: Meta-data was not loaded, because the file was not found: "+str(fnfe))
+        print("I will now proceed to reading an image data.")
+    image_arr = read_binary(in_file, width, height, frames, bytes_per_pix, is_signed, is_float, byte_order)
+    if_success = write_dicom(out_file, image_arr, bytes_per_pix, is_signed, is_float, meta_arr)
+    return if_success
+
+if __name__ == "__main__":
```
Comment (wkrzemien) [inline binary2DICOM.py]: probably the question of taste but I prefer to have separately defined:
def main(argv):
....
if __name__ == "__main__":
 main(sys.argv)

File: binary2DICOM.py (hunk 1 above)
Comment (Rav2) [inline binary2DICOM.py]: Thanks! That's a very useful module.

**Pull request merged**